app/src/main/java/org/astraea/partitioner/partitionerFactory/LinkPartitioner.java

```diff
@@ -0,0 +1,121 @@
+package org.astraea.partitioner.partitionerFactory;
+
+import java.net.MalformedURLException;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Random;
+import org.apache.kafka.clients.producer.Partitioner;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.common.Cluster;
+import org.apache.kafka.common.PartitionInfo;
+import org.astraea.partitioner.nodeLoadMetric.BrokersWeight;
+import org.astraea.partitioner.nodeLoadMetric.LoadPoisson;
+import org.astraea.partitioner.nodeLoadMetric.NodeLoadClient;
+import org.astraea.partitioner.nodeLoadMetric.SingleThreadPool;
+
+public class LinkPartitioner implements Partitioner {
+
+  private static final SmoothPartitionerFactory FACTORY =
+      new SmoothPartitionerFactory(
+          Comparator.comparing(o -> o.get(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG).toString()));
+
+  private Partitioner partitioner;
+
+  @Override
+  public int partition(
+      String topic, Object key, byte[] keyBytes, Object value, byte[] valueBytes, Cluster cluster) {
+    return partitioner.partition(topic, key, keyBytes, value, valueBytes, cluster);
+  }
+
+  @Override
+  public void onNewBatch(String topic, Cluster cluster, int prevPartition) {
+    partitioner.onNewBatch(topic, cluster, prevPartition);
+  }
+
+  @Override
+  public void close() {
+    partitioner.close();
+  }
+
+  @Override
+  public void configure(Map<String, ?> configs) throws NullPointerException {
+
+    var jmxAddresses = configs.get("jmx_servers");
```

Comment:
` var jmxAddresses = Objects.requireNonNull(configs.get("jmx_servers"), "You must configure jmx_servers correctly");`